src/components/ContactsAdd.jsx

```diff
@@ -0,0 +1,156 @@
+//react
+import { useState, useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { useLocation } from 'react-router';
+//helpers
+import urlPath from '../helpers/helpers';
+
+//notes
+// ---useLocation:
+// This hook returns the location object used by the react-router. This object represents the current URL and is immutable. Whenever the URL changes, the useLocation() hook returns a newly updated location object.
```

Comment:
very good idea to add notes like this to revise in the future, i need to do this more often myself!